### PR TITLE
[v1.1][W17-1] tag related documentation changes

### PR DESCRIPTION
**Pull request by ongspxm**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -989,7 +1063,7 @@ Sets of tags for each Section of a resume - to be used to custom generate a resu
 A plaintext file containing a Configuration
 
 [[tag]] Tag::
-A keyword
+A single-word keyword starting with either `#` or `~`
```
Comment (ongspxm) [inline docs/DeveloperGuide.adoc]: adding it here, so that when @jhengy does his "adding" thingy later, he can skip the explanation of tags

Comment (anubh-v) [reply docs/DeveloperGuide.adoc]: Thanks. Just to clarify - the words starting with # (#work, #project) are meant to be categories, not tags.
Each Entry will belong to exactly 1 category.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -936,6 +935,81 @@ Use case ends.
 +
 Use case ends.
 
+[discrete]
+=== Adding tag to an entry
+
+*MSS*
+
+1. User launches application
```
Comment (anubh-v) [inline docs/DeveloperGuide.adoc]: Don't need to say "User launches application" - this was Amrut's feedback on our v1.0 submission

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -936,6 +935,81 @@ Use case ends.
...
+Use case ends.
+
+[discrete]
+=== Viewing all active tags in resume generation
+
+*MSS*
+
+1. User launches application
+2. User enters command to list all tags
+3. System displays all active tags and their entries' placement in the resume.
```
Comment (anubh-v) [inline docs/DeveloperGuide.adoc]: Don't quite understand the meaning of "entries placement in the resume"
Does it mean that the categories of the entries are shown?

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (anubh-v) [reply docs/DeveloperGuide.adoc]: Sorry, my mistake: Words starting with ~ are categories and those starting with # are tags.
See our Google Docs for v1.0 for examples

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (ongspxm) [reply docs/DeveloperGuide.adoc]: like the UI will print out something along this effect to show that the resume will show all the work first (display only java and python related), then the next section will show the projects (only web projects)

```
~work
#java #python

~project
#web
```

**Comment by ongspxm** [comment]
someone merge pls

**Pull request merged**